### PR TITLE
Create AccountDropdown which uses material-ui

### PR DESCRIPTION
**Pull request by befitsandpiper**
update TransferGrain to use new dropdown

Test Plan: manual frontend testing

### CODE REVIEW

File: sharness/__snapshots__/test-instance/data/ledger.json [hunk 1]
```diff
@@ -5,4 +5,4 @@
 {"action":{"identityId":"mbzRr0D3A2wcD1E7p0H2rQ","type":"TOGGLE_ACTIVATION"},"ledgerTimestamp":1595960684984,"uuid":"lXJfGAKpX2C7eV7WQ13I8A","version":"1"},
```
Comment (decentralion) [inline sharness/__snapshots__/test-instance/data/ledger.json:5]: Irrelevant change, please revert

File: src/ui/components/TransferGrain.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ import {
 } from "../../ledger/grain";
 import {type Identity} from "../../ledger/identity";
 import {Ledger, type Account} from "../../ledger/ledger";
-import AccountDropdown from "./AccountSelector";
+import AccountDropdown from "./AccountDropdown";
```
Comment (decentralion) [inline src/ui/components/TransferGrain.js:14]: Looks like we added AccountDropdown without removing AccountSelector. Is there a reason we need both?

File: src/ui/components/TransferGrain.js [hunk 3]
```diff
@@ -25,13 +25,13 @@ export const TransferGrain = ({ledger, setLedger}: Props) => {
   const [maxAmount, setMaxAmount] = useState<Grain>(fromInteger(0));
   const [memo, setMemo] = useState<string>("");
 
-  const setSender = (acct: Account) => {
-    setMaxAmount(acct.balance);
-    setSourceIdentity(acct.identity);
+  const setSender = (acct: Account | null) => {
```
Comment (decentralion) [inline src/ui/components/TransferGrain.js:28]: Why the change in signature?

File: src/ui/components/TransferGrain.js (hunk 2 above)
Comment (befitsandpiper) [reply src/ui/components/TransferGrain.js:14]: It is used on other pages still. But I could make sure everything works with those pages too. Just figured that would lead to styling pages on other pages and this branch quickly getting out of scope.

File: src/ui/components/TransferGrain.js (hunk 3 above)
Comment (befitsandpiper) [reply src/ui/components/TransferGrain.js:28]: I felt it was necessary to be able to set the account back to the null default if you cleared the combobox field. This disables the grain transfer button again and prevents unexpected behavior.

File: src/ui/components/TransferGrain.js (hunk 2 above)
Comment (decentralion) [reply src/ui/components/TransferGrain.js:14]: Yeah, let's update in both places please. I think the PR will still be small & easy to review, and that way we guarantee we aren't adding debt by having two components that do nearly the same thing. It also "validates" the new component a bit by showing that we can re-use it.

Comment (befitsandpiper) [reply src/ui/components/TransferGrain.js]: Okay, I am using this in both places now. This did change how the SpecialGrainDistribution page looks a bit, but did not affect usability, and there will be styling updates there as well in the future. I didn't want to include css changes in this commit.

File: src/ui/components/TransferGrain.js (hunk 3 above)
Comment (decentralion) [reply src/ui/components/TransferGrain.js:28]: Cool, makes sense.

**Comment by befitsandpiper** [comment]
it is full width rn because I haven't applied new styling. This is just switching to material-ui. I have improved styling as well for another commit.
looks more like this:
Screenshot from 2020-08-04 19-50-30

File: src/ui/components/AccountSelector.js [hunk 4]
```diff
@@ -1,102 +1,52 @@
...
-              {item.identity.name}
-            </li>
-          ))}
-      </ul>
-    </div>
+    <ThemeProvider theme={theme}>
+      <Autocomplete
+        onChange={onComboChange}
+        fullWidth
+        id="combo-box-demo"
```
Comment (decentralion) [inline src/ui/components/AccountSelector.js:40]: nit: the id looks vestigial

Comment (befitsandpiper) [reply src/ui/components/AccountSelector.js:40]: ah yes, oops
